src/drivers/gitlab.js

```diff
@@ -162,7 +166,7 @@ class Gitlab {
         --token "${token}" \
         --wait-timeout ${idleTimeout} \
         --executor "${IN_DOCKER ? 'shell' : 'docker'}" \
-        --docker-image "dvcorg/cml:latest" \
+        --docker-image "iterativeai/cml:${gpu ? 'latest-gpu' : 'latest'}" \
```

Comment:
```js
let gpu = true;
    try {
      await exec('nvidia-smi');
    } catch (err) {
      gpu = false;
    }
```

GPU is determined using nvidia. If its not available its NEVER going to execute with GPU